component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -325,4 +393,98 @@ private void allowUser(String userName, Map claims, X509Certificate cert,
     protected boolean retryAuthenticationEnabled() {
         return true;
     }
-}
\ No newline at end of file
+
+    private String buildAlternativeNames(X509Certificate cert, AuthenticationContext authenticationContext) throws AuthenticationFailedException {
+
+        List<String> matchedAlternativeNamesList = new ArrayList<>();
+        try {
+            Collection<List<?>> altNames = cert.getSubjectAlternativeNames();
+            if (altNames != null) {
+                for (List item : altNames) {
+                    ASN1InputStream decoder = null;
+                    if (item.toArray()[1] instanceof byte[])
+                        decoder = new ASN1InputStream((byte[]) item.toArray()[1]);
+                    else if (item.toArray()[1] instanceof String) {
+                        Matcher m = alternativeNamesPatternCompiled.matcher((String) item.toArray()[1]);
+                        addMatchStringsToList(m, matchedAlternativeNamesList);
+                    }
+                    if (decoder == null)
+                        continue;
+                    DEREncodable encoded = decoder.readObject();
```

Comment:
can move 413 to 419 to separate private method for clarity

Reply:
done